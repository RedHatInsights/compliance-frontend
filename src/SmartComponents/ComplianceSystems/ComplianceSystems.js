import React from 'react';
import SystemsTable from '../SystemsTable/SystemsTable';
import { ComplianceTabs } from 'PresentationalComponents';
import { PageHeader, PageHeaderTitle, Main } from '@redhat-cloud-services/frontend-components';

const ComplianceSystems = () => {
    const columns = [{
        composed: ['facts.os_release', 'display_name'],
        key: 'display_name',
        title: 'Name',
        props: {
            width: 40
        }
    }, {
        key: 'facts.compliance.profiles',
        title: 'Profiles',
        props: {
            width: 40
        }
    }, {
        key: 'facts.compliance.compliance_score',
        title: 'Compliance score',
        props: {
            width: 10
        }
    }, {
        key: 'facts.compliance.last_scanned',
        title: 'Last scanned',
        props: {
            width: 10
        }
    }];

    const beta = window.location.pathname.split('/')[1] === 'beta';

    return (
        <React.Fragment>
            <PageHeader className={ beta ? 'beta-page-header' : 'stable-page-header' } >
                <PageHeaderTitle title="Compliance systems" />
                { !beta && <ComplianceTabs/> }
            </PageHeader>
            <Main>
                <SystemsTable columns={columns} />
            </Main>
        </React.Fragment>
    );
};

export default ComplianceSystems;
