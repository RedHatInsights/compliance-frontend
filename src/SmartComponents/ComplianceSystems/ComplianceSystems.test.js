import toJson from 'enzyme-to-json';

import ComplianceSystems from './ComplianceSystems.js';

describe('ComplianceSystems', () => {
    it('expect to render without error', () => {
        const wrapper = shallow(
            <ComplianceSystems />
        );

        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
