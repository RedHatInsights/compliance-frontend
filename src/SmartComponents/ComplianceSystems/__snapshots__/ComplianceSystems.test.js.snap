// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComplianceSystems expect to render without error 1`] = `
<Fragment>
  <PageHeader>
    <PageHeaderTitle
      title="Compliance"
    />
    <withRouter(Connect(RouterParams)) />
  </PageHeader>
  <Connect(Main)>
    <withApollo(SystemsTable)
      columns={
        Array [
          Object {
            "composed": Array [
              "facts.os_release",
              "display_name",
            ],
            "key": "display_name",
            "props": Object {
              "width": 40,
            },
            "title": "Name",
          },
          Object {
            "key": "facts.compliance.profiles",
            "props": Object {
              "width": 40,
            },
            "title": "Profiles",
          },
          Object {
            "key": "facts.compliance.compliance_score",
            "props": Object {
              "width": 10,
            },
            "title": "Compliance score",
          },
          Object {
            "key": "facts.compliance.last_scanned",
            "props": Object {
              "width": 10,
            },
            "title": "Last scanned",
          },
        ]
      }
    />
  </Connect(Main)>
</Fragment>
`;
